feat(shared-data): add Eppendorf skirted 96 well PCR labware definition

theosanderson opened the pull request:
Created from technical drawing at https://www.labscoop.com/p/Eppendorf-Eppendorf-twin-tec-PCR-Plate-96-semi-skirted-250-%C2%B5L-colorless-25-pcs/3887385 and verified with test file.



┆Issue is synchronized with this [Wrike Task](https://www.wrike.com/open.htm?id=535451320) by [Unito](https://www.unito.io/learn-more)

theosanderson committed:
Add 96 well skirted plate, created with labware creator from technical drawing at https://www.labscoop.com/p/Eppendorf-Eppendorf-twin-tec-PCR-Plate-96-semi-skirted-250-%C2%B5L-colorless-25-pcs/3887385

mcous commented on shared-data/labware/definitions/2/eppendorf_twintec_96_wellplate_skirted_pcr/1.json:
```diff
@@ -0,0 +1,1135 @@
+        }
+    ],
+    "parameters": {
+        "format": "irregular",
```
Should be `96Standard` (related to #5085)

mcous commented on shared-data/labware/definitions/2/eppendorf_twintec_96_wellplate_skirted_pcr/1.json:
```diff
@@ -0,0 +1,1135 @@
+        "quirks": [],
+        "isTiprack": false,
+        "isMagneticModuleCompatible": false,
+        "loadName": "eppendorf_twintec_skirted_96"
```
`loadName` needs to match filename and needs to match Opentrons conventions, including:

- Brand name
- Well count
- Labware category
- Nominal well volume
- Modifiers

Not sure what we'd want to do with "Eppendorf twin.tec" as the brand name, though, seeing as in our loadname conventions underscores are section separators. Might not be the biggest deal. Regardless, though, we need to add nominal volume, e.g.

```
eppendorf_twintec_96_wellplate_150ul_pcr_skirted
```

mcous commented on shared-data/labware/definitions/2/eppendorf_twintec_96_wellplate_skirted_pcr/1.json:
```diff
@@ -0,0 +1,1135 @@
+        "isMagneticModuleCompatible": false,
+        "loadName": "eppendorf_twintec_skirted_96"
+    },
+    "namespace": "custom_beta",
```
`namespace` should be `opentrons` in the standard library

mcous reviewed (changes requested):
Thanks for the submission! This may take a while to merge because we have a policy of verifying labware definitions ourselves with physical labware before inclusion in the standard labware library.

In the mean time, seeing as this definition was created using the labware creator, it needs a few modifications before it can be included in the standard library, so I've marked those

theosanderson committed:
Update format to 96Standard

theosanderson committed:
Update loadname

theosanderson committed:
Rename shared-data/labware/definitions/2/eppendorf_twintec_96_wellplate_skirted_pcr/1.json to shared-data/labware/definitions/2/eppendorf_twintec_96_wellplate_150ul_pcr_skirted/1.json

theosanderson committed:
Update 1.json

theosanderson commented:
Thanks, I wasn't quite sure of the conventions. I've addressed these points. 

I do wonder if there is room for some sort of "community" namespace that is clearly listed on the library as not endorsed by OpenTrons. Labware creation is a significant source of friction to trying to do adapt a new protocol to automation, and at the moment there is going to be a lot of duplicated effort. (I'm sure you folks have considered all this a bunch).

theosanderson committed:
Prettier

mcous commented on shared-data/labware/definitions/2/eppendorf_twintec_96_wellplate_150ul_pcr_skirted/1.json:
```diff
@@ -0,0 +1,1023 @@
+  ],
+  "brand": {
+    "brand": "Eppendorf",
+    "brandId": ["0030129512"]
```
The Labware Creator doesn't have UI for this, but we're able to add a `links` field here for the manufacturer's page for this labware. This is the best one I can find: https://online-shop.eppendorf.us/US-en/eshopproduct/view/0030129512.html

mcous commented on shared-data/labware/definitions/2/eppendorf_twintec_96_wellplate_150ul_pcr_skirted/1.json:
```diff
@@ -0,0 +1,1023 @@
+        "displayCategory": "wellPlate",
+        "wellBottomShape": "u"
+      },
+      "brand": {
```
Shoot, did this file come out of the labware creator like this? This nested `brand` field is only supposed to be filled out if the manufacturer of the well group differs from the manufacturer of the labware, such as a tuberack with a certain brand tubes.

This field needs to be removed

theosanderson replied:
Thanks -- pretty sure this came out of the labware creator (for info, not complaint!)

mcous commented on shared-data/labware/definitions/2/eppendorf_twintec_96_wellplate_150ul_pcr_skirted/1.json:
```diff
@@ -0,0 +1,1023 @@
+      "metadata": {
+        "displayName": "Eppendorf twin.tec PCR plate 96 LoBind skirted plate",
+        "displayCategory": "wellPlate",
+        "wellBottomShape": "u"
```
`wellBottomShape` is the only valid field in this `metadata` object; the rest should be removed because this well group is an intrinsic part of the labware itself, and not something external, like tubes.

mcous reviewed (commented):
Shoot, a few more items. No rush or anything because we don't really know how to handle this sort of contribution yet!

theosanderson committed:
Remove well-specific meta

theosanderson committed:
Remove brand from wells

theosanderson committed:
Add link

theosanderson committed:
Update 1.json

theosanderson commented:
Closing for now - considering an alternative way for getting these sorts of contributions available to others.
